pandas/parser.pyx

```diff
@@ -1468,6 +1505,97 @@ cdef _string_box_decode(parser_t *parser, int col,
 
     return result, na_count
 
+@cython.boundscheck(False)
+cdef _categorical_convert(parser_t *parser, int col,
```

Comment:
Well for instance, instead of my fused type hack, just generate two version of `_try_double`, one that reads off the parser, one that takes an input.  Similarly, lines 1533-1548 below are basically generic and repeated in like 7 functions. (`_try_double`, `_try_int64`, `_string_box_decode`, etc)